test/new-e2e/pkg/utils/e2e/stack_definition.go

```diff
@@ -72,29 +73,98 @@ type AgentEnv struct {
 	Agent *client.Agent
 }
 
+// AgentStackDefParam defines the parameters for a stack with a VM and the Datadog Agent
+// installed.
+// The AgentStackDefParam configuration uses the [Functional options pattern].
+//
+// The available options are:
+//   - [WithVMParams]
+//   - [WithAgentParams]
+//   - [WithAgentClientParams]
+//
+// [Functional options pattern]: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
+type AgentStackDefParam struct {
+	vmParams          []ec2params.Option
+	agentClientParams []agentclientparams.Option
+	agentParams       []agentparams.Option
+}
+
+func newAgentStackDefParam(options ...func(*AgentStackDefParam) error) (*AgentStackDefParam, error) {
+	params := &AgentStackDefParam{
+		vmParams:          []ec2params.Option{},
+		agentClientParams: []agentclientparams.Option{},
+		agentParams:       []agentparams.Option{},
+	}
+	for _, o := range options {
+		err := o(params)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return params, nil
+}
+
+// WithVMParams sets VM parameters
+// See [ec2vm.Params] for available options for vmParams.
+func WithVMParams(options ...ec2params.Option) func(*AgentStackDefParam) error {
+	return func(p *AgentStackDefParam) error {
+		p.vmParams = options
```

Comment:
As a user may have several calls to `WithVMParams`, I would use `p.vmParams = append(p.vmParams, options)` to be safe.

Reply:
I followed what was done elsewhere for slices, we replace instead of appending. A dev can pass multiple VM options in one single `WithVMParams` call. 

I think it would be more frequent to have

```golang
AgentStackDef(e2e.WithVMParams(vmParams.WithOS(..), vmParams.WithArch(...), vmParams.WithName(...))
```

rather than

```golang
AgentStackDef(e2e.WithVMParams(vmParams.WithOS(..)), 
                           e2e.WithVMParams(vmParams.WithArch(..)),
                           e2e.WithVMParams(vmParams.WithName(...))
)
```

What do you think ?